rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -114,7 +128,19 @@ impl MultisigCheckpointSyncer {
         // Sort in descending order. The n'th index will represent
         // the highest index for which we (supposedly) have (n+1) signed checkpoints
         latest_indices.sort_by(|a, b| b.cmp(a));
-        if let Some(&highest_quorum_index) = latest_indices.get(threshold - 1) {
+
+        // Find the highest index that meets the threshold weight
+        let mut cumulative_weight = 0;
+        let mut highest_quorum_index = None;
+        for (i, &index) in latest_indices.iter().enumerate() {
+            cumulative_weight += weighted_validators[i].weight; // Add the weight of this validator
```

Comment:
`latest_indices` also isn't guaranteed to have an element for each validator. If the validator didn't sign, there won't be a corresponding element https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/hyperlane-base/src/types/multisig.rs#L88

Reply:
I've fixed it here but with return a tuple of (validator,index) from get_validator_latest_checkpoints_and_update_metrics https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/cc5483565dce6ea71ca0b97f4e6840d4ae8bd700/rust/hyperlane-base/src/types/multisig.rs#L263

Otherwise, just the index is not workable and this function's return isn't needed anywhere else.

Reply:
yeah something like this makes sense, though it'll be more clear to me when I do another review